crates/typst/src/foundations/datetime.rs

```diff
@@ -341,30 +430,52 @@ impl Datetime {
     /// `[[year]-[month]-[day] [hour]:[minute]:[second]]`.
     ///
     /// See the [format syntax]($datetime/#format) for more information.
+    ///
+    /// If you specify `locale`, the `pattern` is ignored, and a different
+    /// algorithm is used to display the datetime in a localized format. You can
+    /// use the `date` and `time` parameters to control the length of the
+    /// output.
     #[func]
     pub fn display(
         &self,
         /// The format used to display the datetime.
         #[default]
         pattern: Smart<DisplayPattern>,
+        /// The locale used to display the datetime. If this is specified,
+        /// `pattern` is ignored and `date` and `time` are used instead.
+        #[named]
+        locale: Option<Str>,
+        /// The length of the localized date. Can be `"full"`, `"long"`,
+        /// `"medium"` or `"short"`. The default is `"medium"`.
+        #[named]
+        #[default(length::Date::Medium)]
+        date: length::Date,
+        /// The length of the localized time. Can be `"medium"` (hour, minute)
+        /// or `"short"` (hour, minute, second).
+        #[named]
+        #[default(length::Time::Medium)]
+        time: length::Time,
     ) -> StrResult<EcoString> {
         let pat = |s| format_description::parse_borrowed::<2>(s).unwrap();
-        let result = match pattern {
-            Smart::Auto => match self {
+        match (pattern, locale) {
```

Comment:
I think `locale` here could be `Smart<Str>`, and when it is `auto`, we could infer the `locale` from `text.lang` & `text.region` (if it is ok).